x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/aggs/heuristic/LongBinomialDistribution.java

```diff
@@ -94,6 +96,68 @@ public double logProbability(long x) {
         return ret;
     }
 
+    public long inverseCumulativeProbability(final double p) throws OutOfRangeException {
+        if (p < 0.0 || p > 1.0) {
+            throw new OutOfRangeException(p, 0, 1);
+        }
+
+        long lower = -1;
+        long upper = numberOfTrials;
+
+        // use the one-sided Chebyshev inequality to narrow the bracket
+        // cf. AbstractRealDistribution.inverseCumulativeProbability(double)
+        final double mu = numberOfTrials * probabilityOfSuccess;
+        final double sigma = FastMath.sqrt(numberOfTrials * probabilityOfSuccess * (1 - probabilityOfSuccess));
+        double k = FastMath.sqrt((1.0 - p) / p);
+        double tmp = mu - k * sigma;
+        if (tmp > lower) {
+            lower = ((int) FastMath.ceil(tmp)) - 1;
+        }
+        k = 1.0 / k;
+        tmp = mu + k * sigma;
+        if (tmp < upper) {
+            upper = ((int) FastMath.ceil(tmp)) - 1;
```

Comment:
I don't know about FastMath so I looked how they implement `ceil` and they do this their own way while HotSpot uses intrinsics, so I suspect that `FastMath.ceil` is actually slower than `Math.ceil`. https://github.com/apache/commons-math/blob/3.6-release/src/main/java/org/apache/commons/math3/util/FastMath.java#L3463